Update README (#178)
Bit-Quill/arrow: cpp/src/arrow/flight/sql/odbc/README.md

      On Windows:
      ```
      .\cpp\src\arrow\flight\sql\odbc\tests\install_odbc.cmd <path\to\repo>\cpp\build\< release | debug >\< Release | Debug>\arrow_flight_sql_odbc.dll
      ```
      Example command for reference:
      ```
      .\cpp\src\arrow\flight\sql\odbc\tests\install_odbc.cmd C:\path\to\arrow\cpp\build\release\Release\arrow_flight_sql_odbc.dll
      ```
      On MacOS:
      ```
      ./cpp/src/arrow/flight/sql/odbc/install/unix/install_odbc.sh <path/to/repo>/cpp/build/< release | debug >/libarrow_flight_sql_odbc.dylib
      ```
      Example command for reference:
      ```
      ./cpp/src/arrow/flight/sql/odbc/install/unix/install_odbc.sh <path/to/arrow>/cpp/build/release/libarrow_flight_sql_odbc.dylib
      ```
      On Linux:
      ```
      ./cpp/src/arrow/flight/sql/odbc/install/unix/install_odbc.sh <path/to/repo>/cpp/build/< release | debug >/libarrow_flight_sql_odbc.so
      ```
      Example command for reference:
      ```
      ./cpp/src/arrow/flight/sql/odbc/install/unix/install_odbc.sh <path/to/arrow>/cpp/build/release/libarrow_flight_sql_odbc.so
      ```

> If the registration is successful, then Apache Arrow Flight SQL ODBC Driver should show as an available ODBC driver. On Windows this should be visible in the x64 ODBC Driver Manager. On MacOS & Linux this should be visible in your odbc.ini.

Reviewer: ```suggestion
If the registration is successful, then Apache Arrow Flight SQL ODBC Driver should show as an available ODBC driver. On Windows this should be visible in the x64 ODBC Driver Manager. On MacOS & Linux this should be visible in your system odbc.ini.
```

      .\cpp\src\arrow\flight\sql\odbc\tests\install_odbc.cmd C:\path\to\arrow\cpp\build\release\Release\arrow_flight_sql_odbc.dll
      ```

If the registration is successful, then Apache Arrow Flight SQL ODBC Driver should show as an available ODBC driver in the x64 ODBC Driver Manager.

## Steps to Register the Apache Arrow Flight SQL ODBC driver on MacOS or Linux

> After the build succeeds, the ODBC .so file will be located in 

Reviewer: ```suggestion
ODBC file is .dylib on macOS and .so on Linux. After the build succeeds, the ODBC file will be located in 
```
Feel free to reword

`build\debug\Debug` for a debug build and `build\release\Release` for a release build.

> 

Reviewer: The ODBC file is `.dylib` on macOS and `.so` on Linux, let's update the docs to mention that

Author: Done.

1. Open terminal.

2. Register your ODBC .so file:

   These commands need `<path\to\repo>` to be replaced with the actual path to the repository.

   1. `cd <path\to\repo>`
   2. Run script to register your ODBC .so file as Apache Arrow Flight SQL ODBC Driver
      ```
      ./cpp/src/arrow/flight/sql/odbc/install/unix/install_odbc.sh <path/to/repo>/cpp/build/< release | debug >/libarrow_flight_sql_odbc.so
      ```
      Example command for reference:
      ```
      ./cpp/src/arrow/flight/sql/odbc/install/unix/install_odbc.sh <path/to/arrow>/cpp/build/release/libarrow_flight_sql_odbc.so

>       ```

Reviewer: Let's add commands for macOS `.dylib` also
```

     ```
      ./cpp/src/arrow/flight/sql/odbc/install/unix/install_odbc.sh <path/to/repo>/cpp/build/< release | debug >/libarrow_flight_sql_odbc.dylib
      ```
      Example command for reference:
      ```
      ./cpp/src/arrow/flight/sql/odbc/install/unix/install_odbc.sh <path/to/arrow>/cpp/build/release/libarrow_flight_sql_odbc.dylib
       ```

```

### Install `.RPM` on Ubuntu
While installing via `.DEB` installer on Ubuntu is the recommended approach, users may install `.RPM` on Ubuntu using below command
```
alien -i --scripts ArrowFlightSqlOdbc-<version>.rpm
```

## Known Limitations

> - Conversion from timestamp data type with specified time zone value to strings is not supported at the moment. This doesn't impact driver's usage of retrieving timestamp data from Power BI on Windows, and Excel on macOS and Windows. See GH-47504 for more context.

Reviewer: right, we already documented this limitation 👍

- Conversion from strings to big int data type has a limit range of -9007199254740992 to 9007199254740992.

> - On Linux, `isql` commands `tables` and `columns` don’t work due to a Dremio backend bug. But users can fetch tables with no issues.

Reviewer: ```suggestion
- On Linux, `isql` commands `tables` and `columns` don’t work due to GH-49702. Users are not blocked from fetching data tables.
```
